File: ShadowFrameworkLite20/src/shadow/geometry/curves/data/SFTransformedCurveData.java
package shadow.geometry.curves.data;

import shadow.geometry.SFCurve;
import shadow.geometry.curves.SFTransformedCurve;
import shadow.math.SFTransform3f;
import shadow.renderer.data.SFDataAsset;
import shadow.renderer.data.SFLibraryReference;
import shadow.renderer.data.transforms.SFNoTransformData;
import shadow.system.data.SFDataCenterListener;
import shadow.system.data.SFNamedParametersObject;

public class SFTransformedCurveData extends SFDataAsset<SFCurve>{

	private SFLibraryReference<SFTransform3f> transform=new SFLibraryReference<SFTransform3f>(new SFNoTransformData());
	private SFLibraryReference<SFCurve> curve =new SFLibraryReference<SFCurve>();
	
	public SFTransformedCurveData() {
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("transform", transform);
		parameters.addObject("curve", curve);
		setData(parameters);
	}
	
	@Override
	protected SFCurve buildResource() {
		final SFTransformedCurve trCurve=new SFTransformedCurve();
		this.transform.retrieveDataset(new SFDataCenterListener<SFDataAsset<SFTransform3f>>() {
			@Override
			public void onDatasetAvailable(String name, SFDataAsset<SFTransform3f> dataset) {
				trCurve.setTrasnform(dataset.getResource());
			}
		});
		this.curve.retrieveDataset(new SFDataCenterListener<SFDataAsset<SFCurve>>() {
			@Override
			public void onDatasetAvailable(String name, SFDataAsset<SFCurve> dataset) {
				trCurve.setCurve(dataset.getResource());
			}
		});
		return trCurve;
	}
	
}

File: ShadowFrameworkJS/src/shadow/geometry/curves/data/SFTransformedCurveData.js


function SFTransformedCurveData(){
	
	this.transform=new SFLibraryReference(new SFNoTransformData());
	this.curve =new SFLibraryReference();
	
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.transform);
	parameters.addObject(this.curve);
	this.setData(parameters);
}

inherit(SFTransformedCurveData,SFDataAsset);

SFTransformedCurveData.prototype["buildResource"]=function(){
	

	var trCurve=new SFTransformedCurve();
	
	var transformDataset=this.transform.retrieveDataset();
	var curveDataset=this.curve.retrieveDataset();
	
	trCurve.setTransform(transformDataset.getResource());
	trCurve.setCurve(curveDataset.getResource());
	
	return trCurve;
};

SFTransformedCurveData.prototype["generateNewDatasetInstance"]=function(){
	return new SFTransformedCurveData();
};	
	



	